webapp/components/post_view/components/post_attachment_opengraph.jsx

```diff
@@ -0,0 +1,192 @@
+    }
+
+    toggleImageVisibility() {
+        this.setState({imageVisible: !this.state.imageVisible});
+    }
+
+    onImageLoad() {
+        this.setState({imageLoaded: true});
+    }
+
+    loadImage(src) {
+        const img = new Image();
+        img.onload = this.onImageLoad;
+        img.src = src;
+    }
+
+    render() {
+        if ($.isEmptyObject(this.state.data) || $.isEmptyObject(this.state.data.description)) {
+            return null;
+        }
+
+        const data = this.state.data;
+        const imageUrl = this.getBestImageUrl();
+        var description = data.description;
+
+        if (description.length > this.maxDescriptionLength) {
+            description = description.substring(0, this.maxDescriptionLength - this.descriptionEllipsis.length) + this.descriptionEllipsis;
+        }
+
+        if (imageUrl && this.state.imageVisible) {
+            this.loadImage(imageUrl);
+        }
+
+        return (
+            <div
+                className='attachment attachment--oembed'
+                ref='attachment'
+            >
+                <div className='attachment__content'>
+                    <div
+                        className={'clearfix attachment__container'}
+                    >
+                        <span className='sitename'>{data.site_name}</span>
+                        <h1
+                            className='attachment__title has-link'
+                        >
+                            <a
+                                className='attachment__title-link'
+                                href={data.url || this.props.link}
+                                target='_blank'
+                                rel='noopener noreferrer'
+                                title={data.title || data.url || this.props.link}
+                            >
+                                {data.title || data.url || this.props.link}
+                            </a>
+                        </h1>
+                        <div >
+                            <div
+                                className={'attachment__body attachment__body--no_thumb'}
+                            >
+                                <div>
+                                    <div>
+                                        {description} &nbsp;
+                                        {imageUrl ?
+                                            <a
+                                                className={'post__embed-visibility'}
+                                                data-expanded={this.state.imageVisible}
+                                                aria-label='Toggle Embed Visibility'
+                                                onClick={this.toggleImageVisibility}
+                                            /> : ''
+                                        }
+                                    </div>
+                                    {imageUrl && this.state.imageVisible ?
```

Comment:
This is minor, but can you not use the tertiary operator for conditionally including JSX and just a regular if statement? I think it's fine to use for the props, but as soon as tertiary operators go multi-line they become much more difficult to read